stacks-signer/src/signerdb.rs

```diff
@@ -988,6 +1023,73 @@ impl SignerDb {
         ))
     }
 
+    /// Get a pending block validation, sorted by the time at which it was added to the pending table.
+    /// If found, remove it from the pending table.
+    pub fn get_and_remove_pending_block_validation(
+        &self,
+    ) -> Result<Option<Sha512Trunc256Sum>, DBError> {
+        let qry = "DELETE FROM block_validations_pending WHERE signer_signature_hash = (SELECT signer_signature_hash FROM block_validations_pending ORDER BY added_time ASC LIMIT 1) RETURNING signer_signature_hash";
```

Comment:
I just realized that this table (and query) could be further simplified. Since the entries are always added in order, there is no need to record the added_time and then order them by that. You can just always remove the oldest entry in the table:
```
DELETE FROM block_validations_pending
WHERE ROWID = (SELECT MIN(ROWID) FROM block_validations_pending)
RETURNING signer_signature_hash;
```

Is there any other use for that `added_time`?

Reply:
This is a pretty good idea, but there isn't actually a "row ID" column - just `signer_signature_hash` and `added_time`, so `added_time` is basically acting as an "ID" here (higher = more recent). So I think the question would be:

- What's the relative performance penalty for using a timestamp vs auto-incrementing ID
- Is there any benefit (either in the future or in debugging) for using a timestamp?

IMO, with these points in mind I'm inclined to say lets just keep the timestamp, but I'm also totally fine with changing it.

Reply:
I agree with @hstove.  Let's leave the timestamp for the time being, in case we later change the insert order into `block_validations_pending`.  Also, row IDs can get reused.

Reply:
Thanks for the discussion. I believe all tables have an implicit ROWID column unless explicitly specified not to (see [here](https://www.sqlite.org/lang_createtable.html#rowid). Regardless, the explanation to keep it sounds good to me.